ui/src/CollectionExerciseDetails.js

```diff
@@ -381,12 +406,103 @@ class CollectionExerciseDetails extends Component {
     this.getActionRules(this.state.authorisedActivities);
   };
 
+  onRescheduleActionRule = async () => {
+    if (this.rescheduleActionRuleDisabled) {
+      return;
+    }
+
+    this.rescheduleActionRuleDisabled = true;
+
+    // Error checking
+    if (!this.state.actionRules.includes(this.state.actionRuleToBeUpdated)) {
+      this.rescheduleActionRuleDisabled = false;
+      return;
+    }
+
+    const actionRule = this.state.actionRuleToBeUpdated;
+    const date = new Date(this.state.updatedTriggerDateTime);
+
+    if (date.getTime() < new Date()) {
+      alert("Trigger time cannot be in the past");
+      this.setState({
+        actionRuleToBeUpdated: {},
+        rescheduleActionRulesDialogDisplayed: false,
+      });
+      return;
+    }
+
+    const dateISOString = date.toISOString();
+
+    if (
+      !confirm(
```

Comment:
Non-critical, but in other places we've used a dialogue to present the confirmation in the same styling as the rest of the page.

Reply:
I've changed it to a dialogue but it has made the code alot messer...